spec/classes/repository_spec.rb

```diff
@@ -9,8 +9,15 @@
       end
 
       context 'with default values for all parameters' do
-        it { is_expected.to compile.with_all_deps }
-        it { is_expected.to contain_class('bareos::repository') }
+        if (facts[:osfamily] == 'Debian') && (facts[:operatingsystemmajrelease] == '11')
+          it 'fails' do
+            expect { is_expected.to contain_class('bareos::repository') }.
+              to raise_error(Puppet::PreformattedError, %r{is not distributed for})
+          end
```

Comment:
Should be more like:

```suggestion
          it { is_expected.to compile.and_raise_error(%r{is not distributed for}) }
```

If you can, put the whole error message to make it obvious what is expected to fail :wink: